src/bson_mapper/bson_archiver.hpp

```diff
@@ -0,0 +1,1041 @@
+#pragma once
+
+#include <memory>
+#include <stack>
+#include <string>
+#include <vector>
+
+#include <cereal/cereal.hpp>
+
+#include <bsoncxx/builder/core.hpp>
+#include <bsoncxx/types.hpp>
+#include <bsoncxx/types/value.hpp>
+
+namespace cereal {
```

Comment:
I missed this before, but I imagine we probably don't want to put our code in cereal's namespace if we're not going to get upstreamed